FrameworkSystem/DB/ProxyDB.py

```diff
@@ -600,13 +643,145 @@ def __getPUSProxy(self, userDN, userGroup, requiredLifetime, requestedVOMSAttr=N
       return S_ERROR('Requested DN does not match the obtained one in the PUSP proxy')
     timeLeft = credDict['secondsLeft']
 
-    result = chain.generateProxyToString(lifeTime=timeLeft,
-                                         diracGroup=userGroup)
+    result = chain.generateProxyToString(timeLeft, diracGroup=userGroup)
     if not result['OK']:
       return result
     proxyString = result['Value']
     return S_OK((proxyString, timeLeft))
 
+  def __getDIRACCAProxy(self, fullName, eMail, DN=None, diracGroup=None, proxyProvider='DIRAC_EOSC_CA'):
+
+    def writeUserConfigFile(fileName, fullName, eMail):
+      userConf = """[ req ]
+        default_bits           = 2048
+        encrypt_key            = yes
+        distinguished_name     = req_dn
+        prompt                 = no
+        req_extensions        = v3_req
+
+        # Generates the following subject
+        # Subject: O=DIRAC CI, O=CERN, CN=ciuser
+        [ req_dn ]
+        C                      = FR
+        O                      = DIRAC
+        OU                     = DIRAC Consortium
+        CN                     = %s
+        emailAddress           = %s
+
+        [ v3_req ]
+        # Extensions for client certificates (`man x509v3_config`).
+        nsComment = "OpenSSL Generated Client Certificate"
+        keyUsage = critical, nonRepudiation, digitalSignature, keyEncipherment
+        extendedKeyUsage = clientAuth
+        """ % (fullName, eMail)
+      with open(fileName, "w") as userConfigFile:
+        userConfigFile.write(userConf)
+
+    if DN:
+      eMail = DN.split('/')[-1].split('=')[1]
+      fullName = DN.split('/')[-2].split('=')[1]
+    caConfigFile = gConfig.getValue('/Resources/ProxyProviders/%s/CAConfigFile' % proxyProvider)
+    maxPeriod = gConfig.getValue('/Resources/ProxyProviders/%s/max_proxylifetime' % proxyProvider) or 365 * 24 * 3600
+    caDir = os.path.dirname(caConfigFile)
+    userConfFile = os.path.join(caDir, fullName.replace(' ', '_') + '.cnf')
+    userReqFile = os.path.join(caDir, fullName.replace(' ', '_') + '.req')
+    userKeyFile = os.path.join(caDir, fullName.replace(' ', '_') + '.key.pem')
+    userCertFile = os.path.join(caDir, fullName.replace(' ', '_') + '.cert.pem')
+    writeUserConfigFile(userConfFile, fullName, eMail)
+    status, output = commands.getstatusoutput('openssl genrsa -out %s 2048' % userKeyFile)
```

Comment:
DIRAC has an internal method used to call commands. But you really needs to use the command? Can  you use  python ssl?